src/main/java/mate/fruitshop/service/FruitService.java

```diff
@@ -0,0 +1,46 @@
+package mate.fruitshop.service;
+
+import static mate.fruitshop.dao.FruitTransactionDaoCsv.FRUIT;
+import static mate.fruitshop.dao.FruitTransactionDaoCsv.QUANTITY;
+import static mate.fruitshop.service.strategy.TransactionHandlerStrategy.TRANSACTIONS_MAP;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import mate.fruitshop.dao.FruitTransactionDao;
+import mate.fruitshop.model.FruitTransaction;
+
+public class FruitService {
+    private FruitTransactionDao dao;
+
+    public FruitService(FruitTransactionDao dao) {
+        this.dao = dao;
+    }
+
+    public Map<String, Integer> calculateFruitsLeft() {
+        List<FruitTransaction> transactions = dao.getAll();
+        Map<String, Integer> fruitQuantityMap = new HashMap<>();
+        int currentFruitQuantity;
+
+        for (FruitTransaction transaction: transactions) {
+            currentFruitQuantity = TRANSACTIONS_MAP.get(transaction.getOperation())
+                    .conductTransaction(transaction,
+                            fruitQuantityMap.getOrDefault(transaction.getFruit(), 0));
+            fruitQuantityMap.put(transaction.getFruit(), currentFruitQuantity);
```

Comment:
```suggestion
            fruitQuantityMap.compute(transaction.getFruit(), (k, v) ->
                    TRANSACTIONS_MAP.get(transaction.getOperation())
                            .conductTransaction(transaction, v == null ? 0 : v));
```
there are a lot of useful methods in hashmap, I think we can try smth like this